### PR TITLE
Avoid issue with deepcopying/pickling IntakeESGFDatasets

### LINKED ISSUE #2989: Recipe run fails when target grid is dataset and data source is intake-esgf
When the `regrid` preprocessor `target_grid` is a dataset and intake-esgf is used as a data source, the run fails with:
```
Traceback (most recent call last):
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_main.py", line 786, in run
    fire.Fire(ESMValTool())
    ~~~~~~~~~^^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/site-packages/fire/core.py", line 135, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/site-packages/fire/core.py", line 468, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
                                ~~~~~~~~~~~~~~~~~~~^
        component,
        ^^^^^^^^^^
    ...<2 lines>...
        treatment='class' if is_class else 'routine',
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        target=component.__name__)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/site-packages/fire/core.py", line 684, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_main.py", line 620, in run
    self._run(recipe, session, cli_config_dir)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_main.py", line 675, in _run
    process_recipe(recipe_file=recipe, session=session)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_main.py", line 153, in process_recipe
    recipe = read_recipe_file(recipe_file, session)
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_recipe/recipe.py", line 92, in read_recipe_file
    return Recipe(raw_recipe, session, recipe_file=filename)
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_recipe/recipe.py", line 922, in __init__
    self.tasks = self.initialize_tasks()
                 ~~~~~~~~~~~~~~~~~~~~~^^
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_recipe/recipe.py", line 1297, in initialize_tasks
    tasks = self._create_tasks()
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_recipe/recipe.py", line 1269, in _create_tasks
    new_tasks, failed = self._create_preprocessor_tasks(
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        diagnostic_name,
        ^^^^^^^^^^^^^^^^
    ...<2 lines>...
        any_diag_script_is_run,
        ^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_recipe/recipe.py", line 1231, in _create_preprocessor_tasks
    task = _get_preprocessor_task(
        datasets=list(datasets),
        profiles=self._preprocessors,
        task_name=task_name,
    )
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_recipe/recipe.py", line 845, in _get_preprocessor_task
    products = _get_preprocessor_products(
        datasets=datasets,
    ...<2 lines>...
        name=task_name,
    )
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/_recipe/recipe.py", line 682, in _get_preprocessor_products
    product = PreprocessorFile(
        filename=filename,
    ...<2 lines>...
        datasets=input_datasets,
    )
  File "/home/bandela/src/esmvalgroup/esmvalcore/esmvalcore/preprocessor/__init__.py", line 647, in __init__
    self.settings = copy.deepcopy(settings)
                    ~~~~~~~~~~~~~^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 163, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 260, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 197, in _deepcopy_list
    append(deepcopy(a, memo))
           ~~~~~~~~^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 163, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 260, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 197, in _deepcopy_list
    append(deepcopy(a, memo))
           ~~~~~~~~^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 163, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 260, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 163, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 260, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 197, in _deepcopy_list
    append(deepcopy(a, memo))
           ~~~~~~~~^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 163, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 260, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 163, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 260, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 163, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 260, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 137, in deepcopy
    y = copier(x, memo)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 222, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ~~~~~~~~^^^^^^^^^^^^^
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/copy.py", line 152, in deepcopy
    rv = reductor(4)
  File "/home/bandela/mambaforge/envs/esmvaltool/lib/python3.13/site-packages/requests_cache/session.py", line 348, in __getstate__
    raise NotImplementedError('CachedSession cannot be pickled')
NotImplementedError: CachedSession cannot be pickled

```

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Closes #2989

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Review by valeriupredoi — APPROVED** [review]
leave the guff about JSON aside for now, adopting the quick solution 🍺 BTW are you familiar with CBOR (since I am on about JSON) - just found out about it - lots smoother than JSON

File: esmvalcore/preprocessor/__init__.py [hunk 1]
```diff
@@ -644,6 +645,22 @@ def __init__(
...
+        # pickled.
+        settings = {
+            fn: {
+                arg: (
+                    value.copy() if is_dataset(value) else copy.deepcopy(value)
+                )
+                for arg, value in kwargs.items()
+            }
+            for fn, kwargs in settings.items()
+        }
```
[the remark this answers is not in the document]
Comment (valeriupredoi) [reply esmvalcore/preprocessor/__init__.py]: yeah that's why I decided to dump (pun intended) my idea above - just realized it's a corner case anyway :grinning:

**Review by schlunma — APPROVED** [review]
I haven't got a suitable recipe to test this. If it works fine for you, please merge it 🚀

**Review by schlunma — APPROVED** [review]
Test ran fine, cheers 🚀
